os/arch/arm/src/amebad/amebad_timer_lowerhalf.c

```diff
@@ -0,0 +1,348 @@
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <tinyara/kmalloc.h>
+#include <tinyara/timer.h>
+#include <tinyara/irq.h>
+#include "timer_api.h"
+
+#ifdef CONFIG_TIMER
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct amebad_gpt_lowerhalf_s {
+	const struct timer_ops_s *ops;	/* Lowerhalf operations */
+	gtimer_t obj;
+	uint8_t irq_id;
+	bool started;				/* True: Timer is started */
+	tccb_t callback;
+	void *arg;					/* Argument passed to upper half callback */
+	uint32_t gpt_timeout;
+};
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+static int amebad_gpt_start(struct timer_lowerhalf_s *lower);
+static int amebad_gpt_stop(struct timer_lowerhalf_s *lower);
+static int amebad_gpt_getstatus(struct timer_lowerhalf_s *lower, struct timer_status_s *status);
+static int amebad_gpt_settimeout(struct timer_lowerhalf_s *lower, uint32_t timeout);
+static void amebad_gpt_setcallback(struct timer_lowerhalf_s *lower, CODE tccb_t callback, void *arg);
+static int amebad_gpt_ioctl(struct timer_lowerhalf_s *lower, int cmd, unsigned long arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+static const struct timer_ops_s g_timer_ops = {
+	.start = amebad_gpt_start,
+	.stop = amebad_gpt_stop,
+	.getstatus = NULL,
+	.settimeout = amebad_gpt_settimeout,
+	.setcallback = amebad_gpt_setcallback,
+	.ioctl = NULL,
+};
+
+static struct amebad_gpt_lowerhalf_s g_gpt0_lowerhalf = {
```

Comment:
What is diffrent each of timer here? for example resolution.

Reply:
Timer 0-3 are the same, they operate with clock source of 32kHz, and the resolution is 32 bits .